### PR TITLE
Add NoGap effect to clear gap pixels

### PR DESCRIPTION
**Pull request by jkbelcher**
This effect can be used on the Master channel to make sure pixel data is never rendered to gap pixels.

### CODE REVIEW

**Comment by jkbelcher** [comment]
Improved, tested, ready for review.  This should be a good tool for finding gap pixel problems and also for suppressing them if needed during showtime.  Changes:

- Collects metrics (% of frames with problems)
- Writes to log on interval (60s default)
- Adjustable interval (long for background, or short for active debugging)
- Writes warning if added to a channel that is not master
- Fast! No looping over pixels.

**Review by raldi — APPROVED** [review]
Crushed it

File: src/main/java/titanicsend/app/TEApp.java [hunk 1]
```diff
@@ -185,9 +185,17 @@ public void initialize(LX lx) {
     lx.registry.addEffect(titanicsend.effect.BasicEffect.class);
     lx.registry.addEffect(titanicsend.effect.Kaleidoscope.class);
     lx.registry.addEffect(titanicsend.effect.NoGapEffect.class);
-    lx.registry.addPatterns(OrganicPatternConfig.getPatterns());
-    lx.registry.addPatterns(ShaderPanelsPatternConfig.getPatterns());
-    lx.registry.addPatterns(ShaderEdgesPatternConfig.getPatterns());
+
+    @SuppressWarnings("unchecked")
```
Comment (raldi) [inline src/main/java/titanicsend/app/TEApp.java:189]: Can we get a comment here?

Comment (jkbelcher) [reply src/main/java/titanicsend/app/TEApp.java:189]: For sure, those are changes from ssilverman's latest commit to main.  I was just resolving merge conflicts to keep this ready to go for you.

Comment (jkbelcher) [reply src/main/java/titanicsend/app/TEApp.java:189]: #152 created the merge conflict that required resolution.

Comment (raldi) [reply src/main/java/titanicsend/app/TEApp.java:189]: Oh, if you're ready to go, you can merge

Comment (jkbelcher) [reply src/main/java/titanicsend/app/TEApp.java:189]: I don't have write access

Comment (raldi) [reply src/main/java/titanicsend/app/TEApp.java:189]: Merged. Also granted you access.

Comment (jkbelcher) [reply src/main/java/titanicsend/app/TEApp.java:189]: It's possible I was doing it wrong.  Anyway, thanks for handling.

**Pull request merged**